### PR TITLE
fix(gitea): fetch both open and closed issues

### PR DESCRIPTION
**Pull request by CirnoT**
- Fetch both closed and open issues
- Explicitly ask for both closed and open PRs (this is default but let's make sure)
- Don't include PRs when asking for issues

### CODE REVIEW

**Comment by rarkins** [comment]
> * Fetch both closed and open issues

Can you clarify what caused you to change this? I ask partly because I want to check we are doing things consistently across all platforms, as I fixed a GitLab issues bug just now too.

**Comment by CirnoT** [comment]
I just noticed this when working on other PR.

By default Gitea returns only open issues, I don't think Renovate cares much whether it finds issue in closed state or doesn't find it (at least as far as ensureClosing goes) but my general understanding of code is that it expects to have list of both types.

File: lib/platform/gitea/gitea-helper.ts [hunk 1]
```diff
@@ -379,7 +379,7 @@ export async function searchIssues(
   params: IssueSearchParams,
   options?: GiteaHttpOptions
 ): Promise<Issue[]> {
-  const query = queryParams(params).toString();
+  const query = queryParams({ ...params, type: 'issue' }).toString();
```
Comment (viceice) [inline lib/platform/gitea/gitea-helper.ts:382]: Why do we need this? Can you link the API docu please.

**Review by viceice — COMMENTED** [review]
@rarkins I think we need this to reopen for example the config error issue.

File: lib/platform/gitea/gitea-helper.ts (hunk 1 above)
Comment (CirnoT) [reply lib/platform/gitea/gitea-helper.ts:382]: https://try.gitea.io/api/swagger#/issue/issueListIssues

```
	// - name: type
	//   in: query
	//   description: filter by type (issues / pulls) if set
	//   type: string
```

```go
	var isPull util.OptionalBool
	switch ctx.Query("type") {
	case "pulls":
		isPull = util.OptionalBoolTrue
	case "issues":
		isPull = util.OptionalBoolFalse
	default:
		isPull = util.OptionalBoolNone
	}
```
`OptionalBoolNone` here means that both are included

Comment (CirnoT) [reply lib/platform/gitea/gitea-helper.ts]: Yes, it's weird that `/issues` API can fetch pulls (and does so by default) but backwards-compatibility...

**Pull request merged**